### ISSUE TITLE
[JSON RPC] Inconsistency between WS and IPC

### ISSUE THREAD

**Issue by dceleda** [ethereum/go-ethereum#23575]
I've tried the same scenario using Web Sockets and IPC and I get different results.

- Send multiple JSON requests in one message. For example: 
{"id": 1, "method": "eth_subscribe", "params": ["newHeads"]}{"id": 1, "method": "eth_subscribe", "params": ["newPendingTransactions"]}
- WS ignores the second request. We can even send smth like this: {"id": 1, "method": "eth_subscribe", "params": ["newHeads"]}{incorrect request}
- IPC handles both requests and returns two responses

**Comment by fjl** [comment]
Yeah, so this is basically the case we discussed: the RPC server does not read beyond the first object in a WebSocket data frame.

**Comment by s1na** [comment]
Yeah exactly and that happens here:

https://github.com/ethereum/go-ethereum/blob/4e599ee469223ccc452f40f36c85bac3bea9c16b/rpc/json.go#L208-L209

Thing is `{message}{message2}` is not valid JSON and the decoder silently ignores the second part. When I feed in a valid JSON message `[{"method: ...}, {"method": ...}]` I get a correct response for both requests.


I personally think how it is right now makes sense, but I could modify the decoder to parse the remainding bytes after decoding the first object. @fjl wdyt?
